### PR TITLE
[AC-1568] Refactor UpdateSecretsManagerSubscriptionCommandTests to use new ctor

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

In https://github.com/bitwarden/server/pull/3121 we added a new, simpler constructor on the SecretsManagerSubscriptionUpdate object, which takes an Organization object and uses it to populate the initial values.

Due to time constraints, that change was backwards-compatible with the old constructor interface. This is part 1 of an effort to go back and update all calling locations to use the new constructor, and then remove the old one.

This PR only updates test code, which was most of the uses of the old ctor. I split it so we can merge the bulk of the changes without needing QA approval. The next PR will make any necessary changes to production code.

While updating the tests I also made some changes to improve their structure and simplify the arrange phase, including:
* autofixture customizations
* ordering tests to match the order of the sut
* using regions

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: test/Core.Test/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommandTests.cs [hunk 1]
```diff
@@ -17,711 +18,617 @@
 namespace Bit.Core.Test.OrganizationFeatures.OrganizationSubscriptionUpdate;
 
 [SutProviderCustomize]
+[SecretsManagerOrganizationCustomize]
 public class UpdateSecretsManagerSubscriptionCommandTests
 {
+    #region General
```
Comment (r-tome) [inline test/Core.Test/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommandTests.cs:24]: We had a vote and `#regions` lost https://contributing.bitwarden.com/contributing/code-style/#code-regions

**Review by r-tome — COMMENTED** [review]
I love the tests, nice work!
Regarding the regions, should we split this file?

File: test/Core.Test/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommandTests.cs (hunk 1 above)
Comment (eliykat) [reply test/Core.Test/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommandTests.cs]: I forgot about that. I removed the regions, they're probably not necessary, and I'm not sure I want to split the test code into separate files.

**Review by r-tome — APPROVED** [review]
LGTM!

**Pull request merged**